plugins/gateway/kong/deploy/deploy.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+KongProxyContainerName="${KONG_PROXY_CONTAINER_NAME:-proxy}"
+KongGatewayDeploymentName="${KONG_GATEWAY_DEPLOYMENT_NAME:-kong}"
+KongGatewayIngressName="${KONG_GATEWAY_INGRESS_NAME:-demo}"
+
+kubectl apply -f kongPlugin.yaml
+
+deploymentPatch=`cat "patch-deployment.yaml" | sed "s/{{KONG_PROXY_CONTAINER_NAME}}/$KongProxyContainerName/g"`
+
+kubectl patch deployments.apps ${KongGatewayDeploymentName} --patch "$deploymentPatch"
+kubectl patch ingresses.networking.k8s.io ${KongGatewayIngressName} --patch "$(cat patch-ingress.yaml)"
```

Comment:
Correct. should we get namespace for each?

Reply:
does the plugin CRD, deployment and Ingress object can be deployed on different namespaces and still work?

Reply:
https://stackoverflow.com/questions/59844622/ingress-configuration-for-k8s-in-different-namespaces

> Now, Ingress Controller can be deployed in any namespace and is, in fact, usually deployed in a namespace separate from your app services. It can out-of-the-box see Ingress rules in all namespaces in the cluster and will pick them up.
The Ingress rules, however, must reside in the namespace where the app that they configure reside.

So the ingress object will probably have different namespace from the controller.
Regarding the `KongPlugin` I'm not sure - you will need to try because there is also [KongClusterPlugin](https://docs.konghq.com/kubernetes-ingress-controller/2.0.x/references/custom-resources/#kongclusterplugin) so not sure if the plugin need to be deployed with the service namespace or the ingress controller namespace